app/src/main/java/com/example/voyagerx/repository/UserRepository.kt

```diff
@@ -70,13 +81,15 @@ class UserRepository @Inject constructor(
         }
     }
 
-    fun register(email: String, password: String){
-        if(getUserByEmail(email) != null){
-            throw Exception(context.getString(R.string.email_already_registered_error))
-        }
+    suspend fun register(email: String, password: String){
+        withContext(Dispatchers.IO) {
+            if (getUserByEmail(email) != null) {
+                throw Exception(context.getString(R.string.email_already_registered_error))
+            }
 
-        database.userDao().insertNoProfile(email, password)
-        logIn(email, password)
+            database.userDao().insertNoProfile(email, password)
+            logIn(email, password)
+        }
     }
 
     fun getCurrentUser(): User?{
```

Comment:
@VivianLi98, Can I use this function to grab the user details for the profile screen? Sorry to ask probably a dumb question but I am walking through your code and getting acquainted with the database.

Reply:
Not a stupid question at all! There is getCurrUser() function in the UserRepository so you can just get it from there. You do need to inject the UserRepository though, but it's pretty straight forward. Just add a @AndroidEntryPoint and a @Inject annotation. You can take a look at my code to see how I did it.

Reply:
Thank you, @VivianLi98!!